src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParameterInput.js

```diff
@@ -42,24 +34,39 @@ const ScenarioParameterInput = ({ parameterData, parametersState, setParametersS
     return null;
   }
 
-  const props = {
-    parameterData,
-    parametersState,
-    setParametersState: setParametersStateSecure,
-    context,
-    key: parameterData.id,
-  };
-  // name property helps distinguish React components from factories
-  if ('name' in varTypeFactory) {
-    return React.createElement(varTypeFactory, props);
-  }
-  console.warn("Warning: Factories are now deprecated and won't be supported in the next major version of the webapp");
-  return varTypeFactory.create(t, parameterData, parametersState, setParametersStateSecure, context);
+  return (
+    <Controller
+      name={parameterData.id}
+      render={({ field }) => {
+        const { value: parameterValue, onChange: setRhfValue } = field;
+
+        const setParameterValue = (newValue) => {
+          if (scenarioIdOnMount.current === getCurrentScenarioId()) {
+            setRhfValue(newValue);
+          }
+        };
+
+        const props = {
+          parameterData,
+          context,
+          key: parameterData.id,
+          parameterValue,
+          setParameterValue,
+        };
+        // name property helps distinguish React components from factories
+        if ('name' in varTypeFactory) {
+          return React.createElement(varTypeFactory, props);
+        }
+        console.warn(
+          "Warning: Factories are now deprecated and won't be supported in the next major version of the webapp"
+        );
+        return varTypeFactory.create(t, parameterData, context, parameterValue, setParameterValue);
```

Comment:
Same remark: we could probably drop support of "factories-as-functions", they have been deprecated since v3.0

Reply:
done